ThingIFSDK/ThingIFSDK/Trigger.swift

```diff
@@ -354,3 +415,80 @@ public class StatePredicate: Predicate {
+    // MARK: - Implements NSCoding protocol
+    public func encodeWithCoder(aCoder: NSCoder) {
+        aCoder.encodeObject(self.endpoint, forKey: "endpoint")
+        aCoder.encodeObject(self.executorAccessToken, forKey: "executorAccessToken")
+        aCoder.encodeObject(self.targetAppID, forKey: "targetAppID")
+        aCoder.encodeObject(self.parameters, forKey: "parameters")
+    }
+    
+    // MARK: - Implements NSCoding protocol
+    public required init(coder aDecoder: NSCoder) {
+        self.endpoint = aDecoder.decodeObjectForKey("endpoint") as! String
+        self.executorAccessToken = aDecoder.decodeObjectForKey("executorAccessToken") as! String
+        self.targetAppID = aDecoder.decodeObjectForKey("targetAppID") as? String
+        self.parameters = aDecoder.decodeObjectForKey("parameters") as? Dictionary<String, AnyObject>
+    }
+    
+    /** Endpoint to call on servercode */
+    public var endpoint: String
+    /** This token will be used to call the external appID endpoint */
+    public var executorAccessToken: String
+    /** If provided, servercode endpoint will be called for this appid. Otherwise same appID of trigger is used */
+    public var targetAppID: String?
+    /** Parameters to pass to the servercode function */
+    public var parameters: Dictionary<String, AnyObject>?
+
+    /** Init TriggeredServerCodeResult with necessary attributes
+     
+     - Parameter endpoint: Endpoint to call on servercode
+     - Parameter executorAccessToken: This token will be used to call the external appID endpoint
+     - Parameter targetAppID: If provided, servercode endpoint will be called for this appid. Otherwise same appID of trigger is used
+     - Parameter parameters: Parameters to pass to the servercode function
+     */
+    public init(endpoint: String, executorAccessToken: String, targetAppID: String?, parameters: Dictionary<String, AnyObject>?) {
+        self.endpoint = endpoint
+        self.executorAccessToken = executorAccessToken
+        self.targetAppID = targetAppID
+        self.parameters = parameters
+    }
+
+    public func toNSDictionary() -> NSDictionary {
+        let dict = NSMutableDictionary(dictionary: ["endpoint": self.endpoint, "executorAccessToken": self.executorAccessToken])
+        if self.targetAppID != nil {
+            dict["targetAppID"] = self.targetAppID
+        }
+        if self.parameters != nil {
+            dict["parameters"] = self.parameters
+        }
+        return dict
+    }
+
+    public override func isEqual(object: AnyObject?) -> Bool {
+        guard let aServerCode = object as? ServerCode else{
+            return false
+        }
+        return self.endpoint == aServerCode.endpoint &&
+            self.executorAccessToken == aServerCode.executorAccessToken &&
+            self.targetAppID! == aServerCode.targetAppID! &&
+            NSDictionary(dictionary: self.parameters!).isEqualToDictionary(aServerCode.parameters!)
+        
```

Comment:
please check whether parameters is nil or not first, then check the dictionary content.